src/IdBrokerClient.php

```diff
@@ -43,18 +35,165 @@ public function __construct(
     }
     
     /**
-     * 
+     * Attempt to authenticate using the given credentials, getting back
+     * information about the authenticated user (if the credentials were
+     * acceptable) or null (if unacceptable).
+     *
+     * @param array $config An array key/value pairs for 'username' and
+     *     'password'.
+     * @return array|null An array of user information (if valid), or null.
+     * @throws Exception
+     */
+    public function authenticate(array $config = [])
+    {
+        $result = $this->authenticateInternal($config);
+        $statusCode = (int)$result['statusCode'];
+        
+        if ($statusCode === 200) {
+            return $this->getResultWithoutStatusCode($result);
+        } elseif ($statusCode === 422) {
+            return null;
+        }
+        
+        throw new Exception(
+            $result['message'] ?? 'Unknown error: ' . $statusCode,
+            1490802360
+        );
+    }
+    
+    /**
+     * Create a user with the given information.
+     *
+     * @param array $config An array key/value pairs of attributes for the new
+     *     user.
+     * @return array An array of information about the new user.
+     * @throws Exception
+     */
+    public function createUser(array $config = [])
+    {
+        $result = $this->createUserInternal($config);
+        $statusCode = (int)$result['statusCode'];
+        
+        if ($statusCode === 200) {
+            return $this->getResultWithoutStatusCode($result);
+        }
+        
+        throw new Exception(
+            $result['message'] ?? 'Unknown error: ' . $statusCode,
+            1490802526
+        );
+    }
+    
+    /**
+     * Deactivate a user.
+     *
+     * @param array $config An array with an 'employee_id' entry.
+     * @throws Exception
+     */
+    public function deactivateUser(array $config = [])
```

Comment:
@fillup I'd like to get your opinion about whether I should leave these functions as they are here, or whether it would make sense to change them to actually just have parameters for the specific pieces of information they need (e.g. the employee id for `deactivateUser(...)`) and internally deal with passing it all along as a keyed array.

Originally, I just used the `$config` parameter for them all, since my focus was handling the status codes so that code using this SDK wouldn't have to, but that means a call to deactivate a user looks like this:
```
$client->deactivateUser([
    'employee_id' => $employeeId,
    'active' => 'no',
]);
```
... which feels rather redundant. I'd rather make it look like this:
```
$client->deactivateUser($employeeId);
```
Is it okay if I make that change (and others like it for these methods in `IdBrokerClient`?